src/java/htsjdk/variant/variantcontext/CommonInfo.java

```diff
@@ -166,29 +189,33 @@ public void setLog10PError(double log10PError) {
     // Working with attributes
     //
     // ---------------------------------------------------------------------------------------------------------
+    /** removes all attributes */
     public void clearAttributes() {
         attributes = new HashMap<String, Object>();
     }
 
     /**
+     * return an unmodifiable Map of attributes
      * @return the attribute map
      */
     public Map<String, Object> getAttributes() {
         return Collections.unmodifiableMap(attributes);
     }
 
     // todo -- define common attributes as enum
-
-    public void setAttributes(Map<String, ?> map) {
+    /** clear all attributes, and set 'map' as attributes */
+    public void setAttributes(final Map<String, ?> map) {
         clearAttributes();
         putAttributes(map);
     }
 
-    public void putAttribute(String key, Object value) {
+    /** shortcut of <code>putAttribute(key, value, false)</code> */
+    public void putAttribute(final String key, final Object value) {
         putAttribute(key, value, false);
     }
 
-    public void putAttribute(String key, Object value, boolean allowOverwrites) {
+    /** insert a new attribute, raises an exception if allowOverwrites==true and the attribute exists */
```

Comment:
This isn't right -- the exception is raised if allowOverwrites is **false** and the attribute exists.